Change shd multiplex into normal svc daemon management

rafikc30 opened the pull request:
shd multiplex was introduced to support dynamic graph
configuration to avoid the glustershd process restarted.
The main use case for this was container storage. This
change was brought by bc3694d.

Since we are no longer actively targetting container storage,
It is better to make the shd as just like the other normal
dameons. Which is more easier to manage and maintain.

Change-Id: I711f39bfb36740b476e00c4c9b4a5c6d11f36f76
Signed-off-by: Mohammed Rafi KC <rafi.kavungal@iternity.com>

rafikc30 commented:
/run regression

gluster-ant commented:
0 test(s) failed 


1 test(s) generated core 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t

1 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t

1 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
https://build.gluster.org/job/gh_centos7-regression/1328/

rafikc30 committed:
glusterd/shd: Change SHD multiplex into normal svc daemon

shd multiplex was introduced to support dynamic graph
configuration to avoid the glustershd process restarted.
The main use case for this was container storage. This
change was brought by bc3694d.

Since we are no longer actively targetting container storage,
It is better to make the shd as just like the other normal
dameons. Which is more easier to manage and maintain.

Change-Id: I711f39bfb36740b476e00c4c9b4a5c6d11f36f76
Signed-off-by: Mohammed Rafi KC <rafi.kavungal@iternity.com>

rafikc30 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/00-geo-rep/georep-stderr-hang.t

0 test(s) generated core 


8 test(s) needed retry 
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
./tests/00-geo-rep/georep-stderr-hang.t

6 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
./tests/000-flaky/bugs_distribute_bug-1122443.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/features_lock-migration_lkmigration-set-option.t
https://build.gluster.org/job/gh_centos7-regression/1329/

rafikc30 commented:
/run full regression

gluster-ant commented:
2 test(s) failed 
./tests/basic/afr/afr-anon-inode.t
./tests/bugs/changelog/bug-1321955.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/basic/afr/afr-anon-inode.t
./tests/basic/glusterd/heald.t
./tests/bugs/changelog/bug-1321955.t
./tests/bugs/glusterfs-server/bug-877992.t

rafikc30 commented:
/run full regression

rafikc30 commented:
/run regression

gluster-ant commented:
5 test(s) needed retry 
./tests/basic/glusterd/thin-arbiter-volume-probe.t
./tests/basic/volume-snapshot.t
./tests/bugs/distribute/bug-1125824.t
./tests/bugs/glusterfs-server/bug-877992.t
./tests/bugs/shard/issue-1425.t

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index f4baf71f8..e35ab6dd0 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
```

rafikc30 commented:
/run regression

gluster-ant commented:
0 test(s) failed 


1 test(s) generated core 
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t

6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_distribute_rebal-all-nodes-migrate.t
./tests/000-flaky/basic_ec_ec-quorum-count-partial-failure.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t

5 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_changelog_changelog-snapshot.t
./tests/000-flaky/basic_ec_ec-quorum-count-partial-failure.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_core_multiplex-limit-issue-151.t
https://build.gluster.org/job/gh_centos7-regression/1426/

rafikc30 commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/cli/src/cli-rpc-ops.c b/cli/src/cli-rpc-ops.c
index ab5c54df3..234296fd5 100644
--- a/cli/src/cli-rpc-ops.c
+++ b/cli/src/cli-rpc-ops.c
@@ -906,9 +906,9 @@ xml_output:
                                       replica_count, disperse_count,
                                       redundancy_count, arbiter_count);
 
-        cli_out("Transport-type: %s",
-                ((transport == 0) ? "tcp"
-                                  : (transport == 1) ? "rdma" : "tcp,rdma"));
+        cli_out("Transport-type: %s", ((transport == 0)   ? "tcp"
+                                       : (transport == 1) ? "rdma"
+                                                          : "tcp,rdma"));
         j = 1;
 
         GF_FREE(local->get_vol.volname);
diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index 646ca8dbc..2b31101eb 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5185,7 +5185,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/libglusterfs/src/dict.c b/libglusterfs/src/dict.c
index 679891a44..f77b94ad4 100644
--- a/libglusterfs/src/dict.c
+++ b/libglusterfs/src/dict.c
@@ -16,14 +16,12 @@
 #include <limits.h>
 #include <fnmatch.h>
 
-
 /* dict_t is always initialized with hash_size = 1
  * with this usage it's implemented as a doubly-linked list
  * and the hash calculation done per operation is not necessary.
  * using this macro to delimit blocks related to hash imp */
 #define DICT_LIST_IMP 1
 
-
 #include "glusterfs/dict.h"
 #if !DICT_LIST_IMP
 #define XXH_INLINE_ALL
@@ -35,7 +33,6 @@
 #include "glusterfs/statedump.h"
 #include "glusterfs/libglusterfs-messages.h"
 
-
 struct dict_cmp {
     dict_t *dict;
     gf_boolean_t (*value_ignore)(char *k);
@@ -581,7 +578,7 @@ dict_get(dict_t *this, char *key)
     }
 #if !DICT_LIST_IMP
     return dict_getn(this, key, strlen(key));
-#else   
+#else
     return dict_getn(this, key, 0);
 #endif
 }
diff --git a/libglusterfs/src/gf-io-uring.c b/libglusterfs/src/gf-io-uring.c
index c8309e33b..7262649d6 100644
--- a/libglusterfs/src/gf-io-uring.c
+++ b/libglusterfs/src/gf-io-uring.c
@@ -58,7 +58,10 @@
 #define GF_IO_URING_FLAG_TIMER GF_IO_ID_FLAG_1
 
 /* Helper macro to define names of bits. */
-#define GF_IO_BITNAME(_prefix, _name) { _prefix##_##_name, #_name }
+#define GF_IO_BITNAME(_prefix, _name)                                          \
+    {                                                                          \
+        _prefix##_##_name, #_name                                              \
+    }
 
 /* Structure to keep names of bits. */
 typedef struct _gf_io_uring_bitname {
@@ -169,8 +172,7 @@ gf_io_uring_dump_params(struct io_uring_params *params)
         GF_IO_BITNAME(IORING_SETUP, CLAMP),
         GF_IO_BITNAME(IORING_SETUP, ATTACH_WQ),
         GF_IO_BITNAME(IORING_SETUP, R_DISABLED),
-        {}
-    };
+        {}};
     static gf_io_uring_bitname_t feature_names[] = {
         GF_IO_BITNAME(IORING_FEAT, SINGLE_MMAP),
         GF_IO_BITNAME(IORING_FEAT, NODROP),
@@ -181,8 +183,7 @@ gf_io_uring_dump_params(struct io_uring_params *params)
         GF_IO_BITNAME(IORING_FEAT, POLL_32BITS),
         GF_IO_BITNAME(IORING_FEAT, SQPOLL_NONFIXED),
         GF_IO_BITNAME(IORING_FEAT, EXT_ARG),
-        {}
-    };
+        {}};
 
     char names[256];
 
@@ -238,8 +239,7 @@ gf_io_uring_dump_ops(struct io_uring_probe *probe)
         [IORING_OP_TEE] = "TEE",
         [IORING_OP_SHUTDOWN] = "SHUTDOWN",
         [IORING_OP_RENAMEAT] = "RENAMEAT",
-        [IORING_OP_UNLINKAT] = "UNLINKAT"
-    };
+        [IORING_OP_UNLINKAT] = "UNLINKAT"};
 
     char names[4096];
     char *ptr;
@@ -873,5 +873,4 @@ const gf_io_engine_t gf_io_engine_io_uring = {
     .flush = gf_io_uring_flush,
 
     .cancel = gf_io_uring_cancel,
-    .callback = gf_io_uring_callback
-};
+    .callback = gf_io_uring_callback};
diff --git a/libglusterfs/src/gf-io.c b/libglusterfs/src/gf-io.c
index 033821d72..f7420cbd7 100644
--- a/libglusterfs/src/gf-io.c
+++ b/libglusterfs/src/gf-io.c
@@ -38,9 +38,7 @@ static const gf_io_engine_t *gf_io_engines[] = {
 #ifdef HAVE_IO_URING
     &gf_io_engine_io_uring,
 #endif
-    &gf_io_engine_legacy,
-    NULL
-};
+    &gf_io_engine_legacy, NULL};
 
 /* Allocate an array of objects in a memory mapped area. */
 #define gf_io_alloc_array(_ptr, _size)                                         \
@@ -374,9 +372,8 @@ GF_IO_CBK(gf_io_test_cancel, op, res, static)
 static int32_t
 gf_io_main(uint32_t workers, gf_io_handlers_t *handlers, void *data)
 {
-    static int32_t signals[] = {
-        SIGSEGV, SIGBUS, SIGILL, SIGSYS, SIGFPE, SIGABRT, SIGCONT, 0
-    };
+    static int32_t signals[] = {SIGSEGV, SIGBUS,  SIGILL,  SIGSYS,
+                                SIGFPE,  SIGABRT, SIGCONT, 0};
     gf_io_thread_pool_t pool;
     gf_io_thread_pool_config_t cfg;
     int32_t res;
diff --git a/libglusterfs/src/glusterfs/gf-io.h b/libglusterfs/src/glusterfs/gf-io.h
index 71f22cf58..494beb011 100644
--- a/libglusterfs/src/glusterfs/gf-io.h
+++ b/libglusterfs/src/glusterfs/gf-io.h
@@ -101,14 +101,14 @@ typedef struct _gf_io_callback {
     const char *name;
     const char *file;
     uint32_t line;
-} *gf_io_callback_t;
+} * gf_io_callback_t;
 
 typedef struct _gf_io_async {
     int32_t (*func)(gf_io_op_t *op);
     const char *name;
     const char *file;
     uint32_t line;
-} *gf_io_async_t;
+} * gf_io_async_t;
 
 #define GF_IO_CBK(_name, _op, _res, _args...)                                  \
     static void __gf_io_cbk_##_name(gf_io_op_t *_op, int32_t _res);            \
@@ -116,8 +116,7 @@ typedef struct _gf_io_async {
         .func = __gf_io_cbk_##_name,                                           \
         .name = #_name,                                                        \
         .file = __FILE__,                                                      \
-        .line = __LINE__                                                       \
-    };                                                                         \
+        .line = __LINE__};                                                     \
     static void __gf_io_cbk_##_name(gf_io_op_t *_op, int32_t _res)
 
 #define GF_IO_ASYNC(_name, _op, _args...)                                      \
@@ -126,8 +125,7 @@ typedef struct _gf_io_async {
         .func = __gf_io_async_##_name,                                         \
         .name = #_name,                                                        \
         .file = __FILE__,                                                      \
-        .line = __LINE__                                                       \
-    };                                                                         \
+        .line = __LINE__};                                                     \
     static int32_t __gf_io_async_##_name(gf_io_op_t *_op)
 
 #define GF_IO_CBK_DECLARE(_name) extern const gf_io_callback_t _name
diff --git a/rpc/rpc-transport/socket/src/socket.c b/rpc/rpc-transport/socket/src/socket.c
index 616161e5f..adc96d974 100644
--- a/rpc/rpc-transport/socket/src/socket.c
+++ b/rpc/rpc-transport/socket/src/socket.c
@@ -4094,7 +4094,7 @@ threadid_func(CRYPTO_THREADID *id)
      */
     CRYPTO_THREADID_set_numeric(id, (unsigned long)pthread_self());
 }
-#else /* older openssl */
+#else  /* older openssl */
 static unsigned long
 legacy_threadid_func(void)
 {
@@ -4350,7 +4350,7 @@ ssl_setup_connection_params(rpc_transport_t *this)
                        "DH ciphers are disabled.",
                        dh_param, ERR_error_string(err, NULL));
             }
-#else /* HAVE_OPENSSL_DH_H */
+#else  /* HAVE_OPENSSL_DH_H */
             BIO_free(bio);
             gf_log(this->name, GF_LOG_ERROR, "OpenSSL has no DH support");
 #endif /* HAVE_OPENSSL_DH_H */
@@ -4377,7 +4377,7 @@ ssl_setup_connection_params(rpc_transport_t *this)
                        "ECDH ciphers are disabled.",
                        ec_curve, ERR_error_string(err, NULL));
             }
-#else /* HAVE_OPENSSL_ECDH_H */
+#else  /* HAVE_OPENSSL_ECDH_H */
             gf_log(this->name, GF_LOG_ERROR, "OpenSSL has no ECDH support");
 #endif /* HAVE_OPENSSL_ECDH_H */
         }
diff --git a/xlators/cluster/afr/src/afr-dir-read.c b/xlators/cluster/afr/src/afr-dir-read.c
index a6a347056..78a889cfb 100644
--- a/xlators/cluster/afr/src/afr-dir-read.c
+++ b/xlators/cluster/afr/src/afr-dir-read.c
@@ -232,9 +232,8 @@ afr_readdir_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     }
 
     if (op_ret >= 0)
-        op_ret = afr_readdir_transform_entries(frame, subvol_entries,
-                                               (long)cookie, &entries,
-                                               local->fd);
+        op_ret = afr_readdir_transform_entries(
+            frame, subvol_entries, (long)cookie, &entries, local->fd);
 
     AFR_STACK_UNWIND(readdir, frame, op_ret, op_errno, &entries, xdata);
 
diff --git a/xlators/features/index/src/index.c b/xlators/features/index/src/index.c
index bebe47fda..3e2b4a418 100644
--- a/xlators/features/index/src/index.c
+++ b/xlators/features/index/src/index.c
@@ -2343,7 +2343,8 @@ index_priv_dump(xlator_t *this)
 
     snprintf(key_prefix, GF_DUMP_MAX_BUF_LEN, "%s.%s", this->type, this->name);
     gf_proc_dump_add_section("%s", key_prefix);
-    gf_proc_dump_write("xattrop-pending-count", "%"PRId64, priv->pending_count);
+    gf_proc_dump_write("xattrop-pending-count", "%" PRId64,
+                       priv->pending_count);
 
     return 0;
 }
diff --git a/xlators/mgmt/glusterd/src/glusterd-store.c b/xlators/mgmt/glusterd/src/glusterd-store.c
index a4a2037f2..d9cdc8d97 100644
--- a/xlators/mgmt/glusterd/src/glusterd-store.c
+++ b/xlators/mgmt/glusterd/src/glusterd-store.c
@@ -2799,7 +2799,8 @@ glusterd_store_retrieve_bricks(glusterd_volinfo_t *volinfo)
             cds_list_add_tail(&ta_brickinfo->brick_list, &volinfo->ta_bricks);
             ta_brick_count++;
             if (gf_store_iter_destroy(&iter)) {
-                gf_msg(this->name, GF_LOG_ERROR, 0, GD_MSG_STORE_ITER_DESTROY_FAIL,
+                gf_msg(this->name, GF_LOG_ERROR, 0,
+                       GD_MSG_STORE_ITER_DESTROY_FAIL,
                        "Failed to destroy store iter");
                 ret = -1;
                 goto out;
diff --git a/xlators/mgmt/glusterd/src/glusterd-volume-set.c b/xlators/mgmt/glusterd/src/glusterd-volume-set.c
index 61218fc47..4f6db1b4e 100644
--- a/xlators/mgmt/glusterd/src/glusterd-volume-set.c
+++ b/xlators/mgmt/glusterd/src/glusterd-volume-set.c
@@ -1057,7 +1057,7 @@ struct volopt_map_entry glusterd_volopt_map[] = {
     {.key = "cluster.ensure-durability",
      .voltype = "cluster/replicate",
      .op_version = 3,
-    .type = NO_DOC,
+     .type = NO_DOC,
      .flags = VOLOPT_FLAG_CLIENT_OPT},
     {.key = "cluster.consistent-metadata",
      .voltype = "cluster/replicate",
@@ -3107,7 +3107,7 @@ struct volopt_map_entry glusterd_volopt_map[] = {
 
     {.key = "rebalance.ensure-durability",
      .voltype = "cluster/distribute",
-    .type = NO_DOC,
+     .type = NO_DOC,
      .op_version = GD_OP_VERSION_10_0,
      .flags = VOLOPT_FLAG_CLIENT_OPT},
     {.key = NULL}};
```

gluster-ant commented:
6 test(s) needed retry 
./tests/000-flaky/basic_afr_split-brain-favorite-child-policy.t
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/bugs/glusterd/brick-order-check-add-brick.t
./tests/bugs/quota/bug-1035576.t
./tests/bugs/replicate/do-not-reopen-fd.t

2 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t

mohit84 commented:
@rafikc30 Can you please refresh the patch.
